src/linker/utilities/singularity_utils.py

```diff
@@ -45,6 +54,11 @@ def _run_cmd(diagnostics_dir: Path, cmd: str, config: Optional[Dict[str, str]])
         #   prepended with 'SINGULARITYENV_'
         env_config = {f"SINGULARITYENV_{key}": value for (key, value) in config.items()}
         env_vars.update(env_config)
+    step_vars = {
+        f"SINGULARITYENV_{input.env_var}": json.dumps(input.container_paths)
```

Comment:
Ah I see, here is where you are using JSON. Did `",".join(input.container_paths)` not work? Or is this a kind of future-proofing against paths with commas in them?

Reply:
This thinking is a bit of a holdover from an earlier revision, when I wasn't directly encoding the environment variables and the `config` object was stringified in a particular way that made deserializing using JSON easier.  That being said, it seems a bit more straightforward (to me) when you have a list of something to serialize and deserialize the list object itself rather than join and split it on either side, but either would work fine

Reply:
This is definitely something where I'd like you to design it, or at least co-design it. I put roughly zero thought into my comma-separated approach. Do you know how other tools do this same kind of thing? I know with command-line arguments (e.g. singularity bindings) it is comma-separated, but not sure about environment variables?